CPAC/distortion_correction/utils.py

```diff
@@ -170,29 +170,44 @@ def phase_encode(unwarp_dir, phase_one, phase_two, dwell_time_one,
 
     """
 
+    meta_data = [dwell_time_one, dwell_time_two,
+                 ro_time_one, ro_time_two]
+    if not any(meta_data):
+        raise Exception("\n[!] Blip-FSL-TOPUP workflow: neither "
+                        "TotalReadoutTime nor DwellTime is present in the "
+                        "epi field map meta-data.")
+
     # create text file
     acq_params = os.path.join(os.getcwd(), "acqparams.txt")
 
     if isinstance(unwarp_dir, bytes):
         unwarp_dir = unwarp_dir.decode()
 
     if unwarp_dir in ["x", "x-", "-x","i","-i","i-"]:
-        dim = nibabel.load(phase_one).shape[0]
-        n_PE_steps = dim - 1
-        ro_time_one = np.round(dwell_time_one * n_PE_steps, 6)
-        ro_time_two = np.round(dwell_time_two * n_PE_steps, 6)
-        ro_times = [f"-1 0 0 {ro_time_one}", f"1 0 0 {ro_time_two}"]
+        if dwell_time_one and dwell_time_two:
+            dim = nibabel.load(phase_one).shape[0]
+            n_PE_steps = dim - 1
+            ro_time_one = np.round(dwell_time_one * n_PE_steps, 6)
+            ro_time_two = np.round(dwell_time_two * n_PE_steps, 6)
+        elif ro_time_one and ro_time_two:
+            ro_times = [f"-1 0 0 {ro_time_one}", f"1 0 0 {ro_time_two}"]
+        else:
+            raise Exception("[!] No dwell time or total readout time "
+                            "present for the acq-fMRI EPI field maps.")
```

Comment:
I think
```suggestion
            raise KeyError("[!] No dwell time or total readout time "
                           "present for the acq-fMRI EPI field maps.")
```
would be appropriate, but maybe `LookupError` or `FileNotFoundError` if it's not necessarily the metadata key that's missing